src/commands/sourcemaps/git.ts

```diff
@@ -0,0 +1,242 @@
+import fs from 'fs'
+import * as simpleGit from 'simple-git'
+import {Writable} from 'stream'
+import {URL} from 'url'
+import {promisify} from 'util'
+
+import {renderGitError, renderProjectPathNotFoundWarning, renderSourcesNotFoundWarning} from './renderer'
+
+// Returns a configured SimpleGit.
+export const newSimpleGit = (): simpleGit.SimpleGit => {
+  const options: simpleGit.SimpleGitOptions = {
+    baseDir: process.cwd(),
+    binary: 'git',
+    // We are invoking at most 3 git commands at the same time.
+    maxConcurrentProcesses: 3,
+  }
+
+  return simpleGit.gitP(options)
+}
+
+// Returns the remote of the current repository.
+export const gitRemote = async (git: simpleGit.SimpleGit): Promise<string> => {
+  const remotes = await git.getRemotes(true)
+  if (remotes.length === 0) {
+    throw new Error('No git remotes available')
+  }
+
+  for (const remote of remotes) {
+    // We're trying to pick the remote called with the default git name 'origin'.
+    if (remote.name === 'origin') {
+      return remote.refs.push
+    }
+  }
+
+  // Falling back to picking the first remote in the list if 'origin' is not found.
+  return remotes[0].refs.push
```

Comment:
Is there a particular reason to use `remotes[0].refs.push` and not `remotes[0].refs.fetch`?

Reply:
None :)